### ISSUE TITLE
Cleanup in data read/write functions in SolverInterface

### ISSUE THREAD

**Issue by kanishkbh** [precice/precice#1525]
I noticed minor issues here and there in the API. Creating this thread to discuss them (and add more as they come up).

### 1. Dimensions and Vertex Count in `writeScalarData(...)`

In  `writeScalarData(...)`, we first check that the dimensions of the writeContext is 1. Then we divide by 1 here:
```C++
const auto vertexCount = values.size() / context.getDataDimensions();
```
Since it's scalar, `vertexCount` would simply be `values.size()`. 

### 2. `values` and `value`

In most of the data read and write functions, `values` represents the input to the function, and `value` represents the internal data structure where the data is written to / read from. This naming is confusing.
Instead of `value`, something like `valuesInternal` would be a better name (as used in `readBlockVectorDataImpl(...)` ).

### 3. `valueIndex` checks

We have
```C++
PRECICE_CHECK(valueIndex >= -1,
```
followed a few lines later by
```c++
PRECICE_CHECK(0 <= valueIndex && valueIndex < vertexCount,
```
And `valueIndex` doesn't change anywhere in these functions. Then the first check looks redundant.

**Comment by MarkAshraf96** [comment]
I am interested in solving this issue.
@kanishkbh  have you already worked on it?

**Comment by fsimonis** [comment]
@kanishkbh Quick feedback:

1. Correct. That's probably copy pasta
2. There is currently a lot changing regarding these data structures. Best no to touch them at the moment.
3. It's best to use [`mesh::Mesh::isVertexIDValid()`](https://precice.org/doxygen/develop/classprecice_1_1mesh_1_1Mesh.html#ac21407160a09623491d9f6364c0a7e43) to check them.

**Comment by fsimonis** [comment]
When composing a couplingscheme [here](https://github.com/precice/precice/pull/1541/files#diff-cb7b50384ae536aad8d8e561e25e510a6f84bd3b95c357c4532a7f27be8e34c4L388-L395), it isn't registered in `_couplingSchemeCompositions`.
So we always compose.

**Comment by kanishkbh** [comment]
> I am interested in solving this issue. @kanishkbh have you already worked on it?

I haven't. Feel free!

**Comment by BenjaminRodenberg** [comment]
What's the status of issue, since #1636 is merged now? I think the situation with value/values/gradientValues is still confusing and could be improved. On the rest I don't have a complete overview.

**Comment by fsimonis** [comment]
I tried to make the naming more consistent. There are still issues though.

This needs to be done before v3.

**Comment by fsimonis** [comment]
Closed by #1636 which got rid of duplications
